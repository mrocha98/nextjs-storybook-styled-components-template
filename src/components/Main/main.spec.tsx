import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import { Main } from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Main />)

    expect(
      screen.getByRole('heading', { name: /template/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
