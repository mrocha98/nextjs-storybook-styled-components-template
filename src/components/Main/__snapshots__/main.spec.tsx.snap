// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  color: #FAFAFA;
  height: 50vh;
  width: 80vh;
  margin: 25vh auto;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 5.6rem;
  margin: 0.8rem 0;
}

.c2 {
  font-size: 2.8rem;
}

.c3 {
  font-size: 2.0rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3.2rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Image of an atom."
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    Template
  </h1>
  <h2
    class="c3"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit.
  </h2>
  <img
    alt="A developer coding in a desktop."
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`
